portal-bridge/src/bridge/history.rs

```diff
@@ -137,25 +136,21 @@ impl HistoryBridge {
                 }
             };
             last_seen_block_counter += 1;
-            if latest_block > block_index {
+            if latest_block >= block_index {
                 last_seen_block_counter = 0;
-                let gossip_range = Range {
-                    start: block_index,
-                    end: latest_block + 1,
-                };
-                info!("Discovered new blocks to gossip: {gossip_range:?}");
-                for height in gossip_range.clone() {
-                    Self::spawn_serve_full_block(
-                        height,
-                        None,
-                        self.portal_clients.clone(),
-                        self.execution_api.clone(),
-                        None,
-                        self.metrics.clone(),
-                    );
-                }
-                block_index = gossip_range.end;
+                info!("Discovered new blocks to gossip: {block_index}-{latest_block}");
+            }
+            for height in block_index..=latest_block {
```

Comment:
This syntax is just a bit unfamiliar to me. Maybe just add a comment explaining what's going on?